jitsi-media-transform/src/main/kotlin/org/jitsi/nlj/PacketInfo.kt

```diff
@@ -24,10 +24,17 @@ import java.time.Clock
 import java.time.Duration
 import java.time.Instant
 import java.util.Collections
+import java.util.concurrent.ConcurrentLinkedQueue
 
 @SuppressFBWarnings("CN_IMPLEMENTS_CLONE_BUT_NOT_CLONEABLE")
 class EventTimeline(
-    private val timeline: MutableList<Pair<String, Duration>> = mutableListOf(),
+    /**
+     * We want this thread safe, because while PacketInfo objects are only handled by a single thread at a time,
+     * [StatsKeepingNode] may add an "exit" event after the packets has been added to another queue potentially handled
+     * by a different thread. This is not critical as it only affects the timeline and the result is just some "exit"
+     * events missing from the trace logs.
+     */
+    private val timeline: ConcurrentLinkedQueue<Pair<String, Duration>> = ConcurrentLinkedQueue(),
```

Comment:
I think `ConcurrentLinkedQueue` is probably more heavyweight than we want for this - it's an actual reader-writer queue, with poll() and the like.

I think Collections.synchronizedList(mutableListOf()), and synchronizing on `timeline` around the `joinToString` (which uses an iterator internally), would be better.